Add Foreman 3.3.1 release notes (#1685)
theforeman/foreman-documentation: guides/doc-Release_Notes/topics/foreman-3.3.1.adoc

* Ubuntu Autoinstall PXELinux template does not reference Smart Proxy correctly - https://projects.theforeman.org/issues/34941[#34941]

=== Compute resources - VMware

* hammer cannot use the cluster name or id as valid input when clusters are residing inside folders and fails with error fog not found - https://projects.theforeman.org/issues/35438[#35438]

=== Tests

* Update show_me_the_cookies to 6.x - https://projects.theforeman.org/issues/35371[#35371]

=== Settings

* Ignore calico interfaces from Foreman - https://projects.theforeman.org/issues/35289[#35289]

=== Host creation

* Bulk rebuild hosts can not reset the build token - https://projects.theforeman.org/issues/35203[#35203]

=== Unattended installations

* Add installation medium for Stream 9 - https://projects.theforeman.org/issues/35151[#35151]

=== Audit Log

* Audit page shows "auditable id / Host2" for "Host1" but Host2 does not exist or deleted from the all hosts - https://projects.theforeman.org/issues/35132[#35132]

=== Inventory

* Hide reports "Origin" column when table is already filtered by origin - https://projects.theforeman.org/issues/35126[#35126]

> * Host\'s power status ping is limited to three seconds - https://projects.theforeman.org/issues/35083[#35083]

Reviewer: ```suggestion
* Host's power status ping is limited to three seconds - https://projects.theforeman.org/issues/35083[#35083]
```

=== Host registration

* In Global Registration, using another LANG like pt_BR.UTF-8 breaks the UUID - https://projects.theforeman.org/issues/35051[#35051]

=== Smart Proxy

> * Smart Proxy upgrade/install fails due to proxy configuration in  \'HTTP(S) proxy\' in settings - https://projects.theforeman.org/issues/34996[#34996]

Reviewer: ```suggestion
* {SmartProxy} upgrade/install fails due to proxy configuration in  \`HTTP(S) proxy\` settings - https://projects.theforeman.org/issues/34996[#34996]
```

=== Web Interface

> * Models index page does 3 same API requests  - https://projects.theforeman.org/issues/34987[#34987]

Reviewer: ```suggestion
* Model's index page does 3 same API requests  - https://projects.theforeman.org/issues/34987[#34987]
```
Not sure about this since I don't know the context but wanted to point this out just in case.

Author: In the navigation it's linked to as Models. If anything you could argue for `Models'` index page.

=== Organizations and Locations

> * Taxonomies are being set during extlogin too early, leading to user\'s default taxonomies being ignored - https://projects.theforeman.org/issues/34949[#34949]

Reviewer: ```suggestion
* Taxonomies are being set during extlogin too early, leading to user's default taxonomies being ignored - https://projects.theforeman.org/issues/34949[#34949]
```

=== Rake tasks

> * \`foreman-rake permissions:reset\` should show a warning that it\'s required to manually update the password in \'foreman.yml\' - https://projects.theforeman.org/issues/31966[#31966]

Reviewer: ```suggestion
* \`foreman-rake permissions:reset\` should show a warning that it's required to manually update the password in \`foreman.yml\` - https://projects.theforeman.org/issues/31966[#31966]
```

== Installer

> 

Reviewer: Is this a higher level title on purpose?

=== External modules

> * SSO feature settings are not getting enabled and failed with HTTPD CONF issue - https://projects.theforeman.org/issues/35524[#35524]

Reviewer: I think this should either use present or past tense but not both.

> * cache-enabled setting for proxy content module isn\'t migrated to \'true\' properly - https://projects.theforeman.org/issues/35157[#35157]

Reviewer: ```suggestion
* cache-enabled setting for proxy content module isn't migrated to \`true\` properly - https://projects.theforeman.org/issues/35157[#35157]
```

== Packaging

> * Sendmail package not present on RHEL8 and needs manual configuration - https://projects.theforeman.org/issues/35461[#35461]

Reviewer: ```suggestion
* \`sendmail\` package is not present on RHEL8 and needs manual configuration - https://projects.theforeman.org/issues/35461[#35461]
```

== Smart Proxy

> 

Reviewer: I would like no note that Smart Proxy section is already included above, unless this is here on purpose as a higher level title.

Author: There is: it reflects the structure in Redmine. So level x.y is the project (https://projects.theforeman.org/projects) and x.y.z is the category. The project Foreman has a category Smart Proxy, so `4.1.10 Smart Proxy` reflects that category. `4.4 Smart Proxy` refers to the Smart Proxy Project. `4.4.1 Packaging` also refers to the Packaging category where `4.3 Packaging` refers to the Packaging project.

Reviewer: Thanks for clarification :+1:

= Foreman 3.3.1

> A full list of changes is available via https://projects.theforeman.org/issues?set_filter=1&sort=id%3Adesc&status_id=closed&f%5B%5D=cf_12&op%5Bcf_12%5D=%3D&v%5Bcf_12%5D%5B%5D=1600[Redmine]

Author: Perhaps you could submit this as a PR against master (against the script) and we could have the discussion there?

== Foreman

=== Templates

> * Restart yggdrasil from kptm if it is already running - https://projects.theforeman.org/issues/35472[#35472]

Author: @adamruzicka looking at it, it doesn't match the content of the patch. It already went live in 3.4.0 so we're late to the party, but it looks like this should be registration. Can you confirm?

> * Restart yggdrasil from kptm if it is already running - https://projects.theforeman.org/issues/35472[#35472]

Author: I think the issue for both me and @evgeni is that we don't really understand the bug is about. There is a very minimal subject and no description. The actual commit doesn't really help me either. Is there a way we could phrase this so it's understandable by users?

> * Restart yggdrasil from kptm if it is already running - https://projects.theforeman.org/issues/35472[#35472]

Reviewer 2: Alright, unless I'm confusing this with another bug in the same area, the original issue was that if you took a fresh machine and registered it in foreman, everything was fine, but if you re-registered it again the machine's entitlement cert got regenerated, but yggdrasil was not restarted so it was still running with the old cert. If you tried to submit a job for that specific host afterwards, the job would be addressed to the new identity and the job would just hang. This was fixed by conditionally restarting yggdrasild when the configuration snippet runs.

The same issue could be triggered using katello-pull-transport-migrate (the rpm) and messing with sub-man by hand. It is a bit hard to categorize the fix as it landed in templates in foreman and katello-pull-transport-migrate, but the bug could be triggered using registration.

I must admit the description in redmine is not great and the description in linked bugzilla has some red herrings in it. Hope what I wrote above is helps shed some light into it

* ks= kernel parameter in Kickstart default iPXE causes RHEL9 Anaconda failure to start - https://projects.theforeman.org/issues/34975[#34975]
* Allow setting up rex pull during provisioning and registration - https://projects.theforeman.org/issues/34969[#34969]

> * Ubuntu Autoinstall PXELinux template does not reference Smart Proxy correctly - https://projects.theforeman.org/issues/34941[#34941]

Author: These release notes are only used in upstream. I'd say we shouldn't try to deal with branding.

=== Compute resources - VMware

* hammer cannot use the cluster name or id as valid input when clusters are residing inside folders and fails with error fog not found - https://projects.theforeman.org/issues/35438[#35438]

=== Tests

> * Update show_me_the_cookies to 6.x - https://projects.theforeman.org/issues/35371[#35371]

Author: It's a dependency. Added that.

=== Settings

* Ignore calico interfaces from Foreman - https://projects.theforeman.org/issues/35289[#35289]
* Fix the SettingValueException error message - https://projects.theforeman.org/issues/34921[#34921]

=== Host creation

* Bulk rebuild hosts can not reset the build token - https://projects.theforeman.org/issues/35203[#35203]

=== Unattended installations

* Add installation medium for Stream 9 - https://projects.theforeman.org/issues/35151[#35151]

=== Audit Log

> * Audit page shows "auditable id / Host2" for "Host1" but Host2 does not exist or deleted from the all hosts - https://projects.theforeman.org/issues/35132[#35132]

Reviewer: ```suggestion
* Audit page shows "auditable id / Host2" for Host1, but Host2 does not exist or deleted from the all hosts - https://projects.theforeman.org/issues/35132[#35132]
```
Also, `Host2 does not exist or deleted from the all hosts` should probably be reworded as I am not really sure what this sentence was meant to say. Maybe "Host2 does not exist or **is** deleted from the all hosts"?

=== Inventory

* Hide reports "Origin" column when table is already filtered by origin - https://projects.theforeman.org/issues/35126[#35126]

> * Hosts power status ping is limited to three secondss power status ping is limited to three seconds - https://projects.theforeman.org/issues/35083[#35083]

Reviewer 3: ```suggestion
* Hosts power status ping is limited to three seconds power status ping is limited to three seconds - https://projects.theforeman.org/issues/35083[#35083]
```

Author: And here

=== Host registration

> * Global Registration, if you use another LANG like pt_BR.UTF-8, it will broken the UUID - https://projects.theforeman.org/issues/35051[#35051]

Reviewer 3: ```suggestion
* In Global Registration, using another LANG like pt_BR.UTF-8 breaks the UUID - https://projects.theforeman.org/issues/35051[#35051]
```

=== Smart Proxy

> * Smart Proxy upgrade/install fails due to proxy configuration in  HTTP(S) proxy' in settingsHTTP(S) proxy in settings in settings - https://projects.theforeman.org/issues/34996[#34996]

Reviewer 3: wat :)

Author: Oh, there's a very weird issue with quotes: here I try to escape it to make it asciidoc safe, but it has "some issues":
https://github.com/theforeman/foreman-documentation/blob/3357580287bebfa749158d3c499ebc4802d5258c/guides/doc-Release_Notes/redmine_release_notes#L64
However, I never really understood why.

Author: Ah, needs more backslashes: https://stackoverflow.com/questions/2180322/ruby-gsub-doesnt-escape-single-quotes

> * Smart Proxy upgrade/install fails due to proxy configuration in  HTTP(S) proxy' in settingsHTTP(S) proxy in settings in settings - https://projects.theforeman.org/issues/34996[#34996]

Reviewer: ```suggestion
* {SmartProxy} upgrade/install fails due to proxy configuration in  HTTP(S) proxy settings - https://projects.theforeman.org/issues/34996[#34996]
```
Maybe the sentence was meant to say this?

Author: Latest version is fixed. Also the same comment as above about branding.

=== Web Interface

* Models index page does 3 same API requests  - https://projects.theforeman.org/issues/34987[#34987]

=== Organizations and Locations

> * Taxonomies are being set during extlogin too early, leading to users default taxonomies being ignoreds default taxonomies being ignored - https://projects.theforeman.org/issues/34949[#34949]

Reviewer: ```suggestion
* Taxonomies are being set during extlogin too early, leading to user's default taxonomies being ignored - https://projects.theforeman.org/issues/34949[#34949]
```

=== Rake tasks

* \`foreman-rake permissions:reset\` should show a warning that its required to manually update the password in 'foreman.yml's required to manually update the password in foreman.yml'foreman.yml - https://projects.theforeman.org/issues/31966[#31966]

== Installer

=== External modules

* SSO feature settings are not getting enabled and failed with HTTPD CONF issue - https://projects.theforeman.org/issues/35524[#35524]

> * cache-enabled setting for proxy content module isnt migrated to 'true' properlyt migrated to true' properlytrue properly properly - https://projects.theforeman.org/issues/35157[#35157]

Reviewer 3: :confused: 
```suggestion
* cache-enabled setting for proxy content module isn't migrated to 'true' properly - https://projects.theforeman.org/issues/35157[#35157]
```

Author: Same here.
